library/common/extensions/filters/http/local_error/filter.cc

```diff
@@ -68,14 +59,16 @@ Http::FilterHeadersStatus LocalErrorFilter::encodeHeaders(Http::ResponseHeaderMa
   }
 
   const auto grpc_status = Grpc::Common::getGrpcStatus(headers);
-  if (grpc_status && grpc_status.value() != Grpc::Status::WellKnownGrpcStatus::Ok) {
+  // gRPC status in headers implies this is an error.
+  if (grpc_status) {
     ASSERT(end_stream,
            "Local gRPC responses must consist of a single headers frame. If Envoy changes "
            "this expectation, this code needs to be updated.");
     ENVOY_LOG(debug, "intercepted local GRPC response");
-    mapLocalGrpcResponseToError(grpc_status.value(), headers);
-    headers_->addCopy(Http::InternalHeaders::get().ErrorMessage,
-                      Grpc::Common::getGrpcMessage(headers));
+    uint64_t http_status = Grpc::Utility::grpcToHttpStatus(grpc_status.value());
```

Comment:
is the local var just for ease of reading?

Reply:
Yep. The compiler will almost certainly ignore it.